wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -710,44 +721,163 @@ public void init(InitRequest request) throws GitException {
+                Revision revision = getRevision(commit, sFilePath);
                 commits.add(revision);
             }
             return new LogPage(commits);
         } catch (GitAPIException | IOException exception) {
-            String errorMessage = exception.getMessage();
-            if (ERROR_LOG_NO_HEAD_EXISTS.equals(errorMessage)) {
-                throw new GitException(errorMessage, ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED);
+			String errorMessage = exception.getMessage();
+			LOG.error("Failed to retrieve log. ", errorMessage);
+			if (ERROR_LOG_NO_HEAD_EXISTS.equals(errorMessage)) {
+				throw new GitException(errorMessage, ErrorCodes.INIT_COMMIT_WAS_NOT_PERFORMED);
+			}else{
+				throw new GitException(exception);
+			}
+        }
+    }
+
+    private Revision getRevision(RevCommit commit, String sFilePath) throws GitAPIException, IOException {
+        List<String> commitParentsList = Stream.of(commit.getParents())
+                                               .map(RevCommit::getName)
+                                               .collect(Collectors.toList());
+        List<Branch> commitBranchesList = getCommitBranches(commit, ListMode.ALL);
+        List<DiffCommitFile> diffCommitFilesList = getCommitDiffFiles(commit,sFilePath);
+
+        GitUser author = getCommitAuthor(commit);
+        GitUser committer = getCommitCommitter(commit);
+
+        Revision revision = createDto(Revision.class).withId(commit.getId()
+                                                     .getName())
+                                                     .withMessage(commit.getFullMessage())
+                                                     .withCommitTime((long) commit.getCommitTime() * 1000)
+                                                     .withCommitter(committer)
+                                                     .withAuthor(author)
+                                                     .withBranches(commitBranchesList)
+                                                     .withCommitParent(commitParentsList)
+                                                     .withDiffCommitFile(diffCommitFilesList);
+        return revision;
+    }
+
+    private GitUser getCommitCommitter(RevCommit commit) {
+        PersonIdent committerIdentity = commit.getCommitterIdent();
+        GitUser committer = createDto(GitUser.class).withName(committerIdentity.getName())
+                                                     .withEmail(committerIdentity.getEmailAddress());
+        return committer;
+    }
+
+    private GitUser getCommitAuthor(RevCommit commit) {
+        PersonIdent authorIdentity = commit.getAuthorIdent();
+        GitUser author = createDto(GitUser.class).withName(authorIdentity.getName())
+                                                  .withEmail(authorIdentity.getEmailAddress());
+        return author;
+    }
+
+    private List<Branch>  getCommitBranches(RevCommit commit, ListMode mode) throws GitAPIException {
```

Comment:
The name of this method is not very clear, may be it would be better to name it `getBranchesOfCommit` ?